src/query/query_parser/query_parser.rs

```diff
@@ -68,6 +69,9 @@ pub enum QueryParserError {
     /// The format for the date field is not RFC 3339 compliant.
     #[error("The date field has an invalid format")]
     DateFormatError(chrono::ParseError),
+    /// The format for the facet field invalid.
+    #[error("The facet field is malformed")]
+    FacetFormatError(String),
```

Comment:
While an additional entry in `TantivyError` for this will possibly be handy later, I prefer this to be a query parser error like others (e.g. date time format error)  because one can think Facet representations are coincidentally the same between Facet and QueryParser.

Reply:
From the view point of `Facet::from_text()`, it will just suffice, for now, to return with a single kind of error (meaning `TantivyError::InvalidArgument`) because the parser isn't made possible to report a detailed error.

Reply:
I had another look and think `from_text` should return a `FacetParseError`. This error would then be included in the tantivty and parse error sets. Currently there is a conversion to `_ => Err(QueryParserError::SyntaxError),`, but this is dead code as I understand it, which would be removed.

```
/// Error that may occur when parsing a facet
#[derive(Debug, Error)]
pub enum FacetParseError {
    /// The format for the facet field invalid.
    #[error("The facet field is malformed")]
    FacetParseError(String),
}
```


```
pub enum TantivyError {
    /// Failed to parse facet.
    #[error("{0}")]
    FacetParseError(#[from] FacetParseError),
```

Reply:
Having a specific error enum for the facet parser totally makes sense to me.

Reply:
I just added an enum to facet as I wouldn't think extending `TantivyError` was necessary.